x/precisebank/genesis.go

```diff
@@ -0,0 +1,37 @@
+package precisebank
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/kava-labs/kava/x/precisebank/keeper"
+	"github.com/kava-labs/kava/x/precisebank/types"
+)
+
+// InitGenesis initializes the store state from a genesis state.
+func InitGenesis(
+	ctx sdk.Context,
+	keeper keeper.Keeper,
+	ak types.AccountKeeper,
+	gs *types.GenesisState,
+) {
+	if err := gs.Validate(); err != nil {
+		panic(fmt.Sprintf("failed to validate %s genesis state: %s", types.ModuleName, err))
+	}
+
+	// initialize module account
+	if moduleAcc := ak.GetModuleAccount(ctx, types.ModuleName); moduleAcc == nil {
+		panic(fmt.Sprintf("%s module account has not been set", types.ModuleName))
+	}
```

Comment:
nit: can we add a test for this similar to the one we have in x/community?

https://github.com/Kava-Labs/kava/blob/3c53e72220377d0522e6aeeaa8ba8cd794d96bbd/x/community/module_test.go#L14-L22